pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/Context.java

```diff
@@ -265,4 +269,5 @@
      */
     <O> CompletableFuture<Void> publish(String topicName, O object, String schemaOrSerdeClassName, Map<String, Object> messageConf);
```

Comment:
Can you add `@Deprecated in favor of using {@link #newOutputMessage}` to all the context publish methods?